prow/pod-utils/gcs/upload.go

```diff
@@ -36,11 +40,33 @@ type UploadFunc func(writer dataWriter) error
 type destToWriter func(dest string) dataWriter
 
 // Upload uploads all of the data in the
-// uploadTargets map to GCS in parallel. The map is
-// keyed on GCS path under the bucket
-func Upload(bucket *storage.BucketHandle, uploadTargets map[string]UploadFunc) error {
+// uploadTargets map to blob storage in parallel. The map is
+// keyed on blob storage path under the bucket
+func Upload(bucket, gcsCredentialsFile, s3CredentialsFile string, uploadTargets map[string]UploadFunc) error {
+	parsedBucket, err := url.Parse(bucket)
+	if err != nil {
+		return fmt.Errorf("cannot parse bucket name %s: %v", bucket, err)
+	}
+	if parsedBucket.Scheme == "" || parsedBucket.Scheme == "gs" {
+		if parsedBucket.Scheme == "gs" {
+			bucket = strings.Trim(bucket, "gs://")
+		}
+		gcsClient, err := storage.NewClient(context.Background(), option.WithCredentialsFile(gcsCredentialsFile))
```

Comment:
This would mean SetSize and ApplyAttributes are not possible anymore because the opener interface does not provide that right now. But I gues that's your point in the comments below

Reply:
sgtm

Reply:
Not sure I understand your comment @fejta , how do we "prove it works" and how does that give us a guarantee that migration GCS over will not result in failures?

I am in favor of migrating it over now, as this is a pretty binary thing: It will work or not, but its unlikely that it will work sometimes. So one way or another, we get very fast feedback.

Reply:
("Now" not necessarily being this PR as its huge and there is a good chance we miss things, but as a follow-up pretty directly after this PR is in and which we can easily revert should we see issues)